src/builder.ts

```diff
@@ -4,12 +4,20 @@ import { getProjectDirectoryPath } from './path.helpers';
 import { loadPackageJson } from './file.helpers';
 import * as rimraf from 'rimraf';
 import { promisify } from 'util';
-
-const execAsync = promisify(exec);
-const rimrafAsync = promisify(rimraf);
+import { PackageJson } from './interfaces';
 
 export class Builder {
 
+  public static create(): Builder {
+    return new Builder(loadPackageJson, promisify(rimraf), promisify(exec));
+  }
+
+  constructor (
+    private loadPackageJson: () => PackageJson,
+    private rimraf: (path: string) => Promise<void>,
+    private exec: (command: string) => Promise<{ stdout: string, stderr: string }>
+  ) {}
```

Comment:
Just out of curiosity: which advantage do you see to have these function as injected parameters? 

The only thing that I can think of it's an easier mocking strategy for testing the `Builder` class, but I'm not sure what opinion I have about this. I'm not sure about the idea of having to adapt my implementation for testing reasons, so that the mocking is easier. For testing imported classes, `jest` provides already a toolset to do this. But maybe it's for another reason that I'm not seeing :)